app/src/lib/shell.ts

```diff
@@ -0,0 +1,107 @@
+/* tslint:disable:no-sync-functions */
+
+import { spawnSync } from 'child_process'
+import * as os from 'os'
+
+const ENVIRONMENT_VARIABLES_TO_PRESERVE = new Set([ 'NODE_ENV', 'NODE_PATH' ])
+
+type IndexLookup = {
+  [propName: string]: string;
+}
+
+/**
+ * Inspect whether the current process is being launched from the macOS tray,
```

Comment:
It's not just if it's launched from the tray/Dock, it's basically if it's launched from anywhere a user would launch it.